connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -274,18 +276,23 @@ public static NewTopicBuilder defineTopic(String topicName) {
      * @param adminConfig the configuration for the {@link Admin}
      */
     public TopicAdmin(Map<String, Object> adminConfig) {
-        this(adminConfig, Admin.create(adminConfig));
+        this(adminConfig.get(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG), Admin.create(adminConfig));
     }
 
-    // visible for testing
-    TopicAdmin(Map<String, Object> adminConfig, Admin adminClient) {
-        this(adminConfig, adminClient, true);
+    /**
+     * Create a new topic admin using the provided {@link Admin}
+     *
+     * @param bootstrapServers the Kafka cluster targeted by the admin
+     * @param adminClient the {@link Admin} to use under the hood
+     */
+    public TopicAdmin(Object bootstrapServers, Admin adminClient) {
```

Comment:
This constructor kinda confuses the ownership of the `Admin` client. I think things are cleaner when the TopicAdmin instantiates (and thus owns the `admin`). Note, it looks like there are no callers for `TopicAdmin.admin`.  It seems that the call sites in `doBuild` could simply pass the map of configs (and the `bootstrapServers` looked up from that), rather than instantiating the `admin` and then passing it to the TopicAdmin.

Obviously the test code has slightly different requirements, meaning we still need this constructor. I did also wonder whether we could also get rid of `bootstrapServers` by defining `toString` on `KafkaAdminClient` and using that for the logging and exceptions here in `TopicAdmin`. Perhaps that's worth a followup PR at some point, (though perhaps there are benefits to hiding bootstrap servers from receivers of clients).

Reply:
Good points. I've:

- Demoted the visibility of the constructor that accepts an `Admin` instance back to package-private
- Documented that it's the caller's responsibility to close the topic admin in order to release the resources allocated for its `Admin`
- Removed the unused `admin` method
- Switched over to the `TopicAdmin(Map<String, Object>)` constructor in the `Worker` class

I originally did this refactoring because I was managing the lifecycle of the `Admin` differently, but kept part of it because it made testing much easier when all you had to pass in was an `Admin` and a `String` instead of an `Admin` and an entire config map. It also seemed like an anti-pattern to have a constructor that accepted both an already-instantiated client object, and the configuration for that client object.